lib/matplotlib/patches.py

```diff
@@ -3330,6 +3394,22 @@ def transmute(self, path, mutation_size, linewidth):
                 else:
                     _path.append(Path(verticesA, codesA))
                     _fillable.append(False)
+            elif self.beginarrow_bracket:
+                vertices_list, codes_list = [], []
+                x0, y0 = path.vertices[0]
+                x1, y1 = path.vertices[1]
+                cos_t, sin_t = get_cos_sin(x1, y1, x0, y0)
```

Comment:
It looks like the `get_cos_sin` call is the same for both begin and end; can it go directly in `_get_bracket`?